src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/PortableInfobox.java

```diff
@@ -225,177 +223,124 @@ public PortableInfobox open(String articleTitle) {
     return this;
   }
 
-  public PortableInfobox compareURLAndExternalLink(String externalLinkName,
-                                                   String externalNavigatedURL) {
-    Assertion.assertEquals(externalLinkName, externalNavigatedURL);
-
-    return this;
+  public String getHorizontalItemLabelFontSize(int index) {
+    return horizontalItemLabels.get(index).getCssValue("font-size");
   }
 
-  public void compareURLAndInternalLink(String internalLinkName, String internalNavigatedURL) {
-    Assertion.assertEquals(internalLinkName, internalNavigatedURL);
+  public String getHorizontalItemValuesFontSize(int index) {
+    return horizontalItemValues.get(index).getCssValue("font-size");
   }
 
-  public PortableInfobox compareFontSizes(WebElement firstElement,
-                                          WebElement secondElement) {
-    Assertion.assertEquals(
-        firstElement.getCssValue("font-size"),
-        secondElement.getCssValue("font-size")
-    );
-
-    return this;
+  public String getItemValuesFontSize(int index) {
+    return itemValues.get(index).getCssValue("font-size");
   }
 
-  public PortableInfobox compareFontSizesBetweenHorizontalItemLabelAndItemLabel() {
-    compareFontSizes(horizontalItemLabel, itemLabels.get(0));
-
-    return this;
+  public String getItemLabelsFontSize(int index) {
+    return itemLabels.get(index).getCssValue("font-size");
   }
 
-  public PortableInfobox compareFontSizesBetweenHorizontalItemValueAndItemValue() {
-    compareFontSizes(horizontalItemValue, itemValues.get(0));
-
-    return this;
+  public String getOrderedElementFontSize(int index) {
+    return orderedElements.get(index).getCssValue("font-size");
   }
 
-  public PortableInfobox compareFontSizesBetweenItemValueAndOrderedListItemWithIndex(int index) {
-    compareFontSizes(itemValues.get(0), orderedElements.get(index));
-
-    return this;
+  public String getUnorderedElementFontSize(int index) {
+    return unorderedElements.get(index).getCssValue("font-size");
   }
 
-  public PortableInfobox compareFontSizesBetweenItemValueAndUnorderedListItemWithIndex(int index) {
-    compareFontSizes(itemValues.get(0), unorderedElements.get(index));
-
-    return this;
+  private boolean isElementVisible(WebElement element) {
+    try {
+      return element.isDisplayed();
+    } catch (NoSuchElementException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)